web/federate_test.go

```diff
@@ -0,0 +1,159 @@
+// Copyright 2016 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package web
+
+import (
+	"bytes"
+	"net/http/httptest"
+	"sort"
+	"strings"
+	"testing"
+
+	"github.com/prometheus/common/model"
+	"github.com/prometheus/prometheus/promql"
+)
+
+var scenarios = map[string]struct {
+	params string
+	accept string
+	code   int
+	body   string
+}{
+	"empty": {
+		params: "",
+		code:   200,
```

Comment:
Yes, indeed out of scope.

But for the record: I read the doc as "if you don't specify at least one `match[]` parameter, you will select zero series and get an empty response" rather than "if you don't specify at least one `match[]` parameter, you will get a 400".

Just hitting a parameter-less endpoint should usually do something sane, IMHO. At least, a @matthiasr set up a health check at a weird company called SoundCloud, where he just hits `/federate`.

So I'd argue the current behavior both makes sense and is in line with the docs (although they could be worded more clearly).